sdk/src/main/java/com/horizen/account/transaction/EthereumTransaction.java

```diff
@@ -35,83 +31,157 @@
 @JsonView(Views.Default.class)
 public class EthereumTransaction extends AccountTransaction<AddressProposition, SignatureSecp256k1> {
 
-    private final RawTransaction transaction;
-    private AddressProposition fromAddress = null;
-    private String id = null;
+    //  The 3 versions of tx are supported by go eth and we have test vectors generated using all of them
+    //  We are using elsewhere the enum from w3j, which just supports 0 and 2:
+    //   org/web3j/crypto/transaction/type/TransactionType.java
+    public enum EthereumTransactionType {
+        LegacyTxType,     // Legacy
+        AccessListTxType, // - not supported
+        DynamicFeeTxType  // eip1559
+    }
+
+    private final EthereumTransactionType type;
+    private final BigInteger nonce;
 
-    private final BigInteger txCost;
+    @JsonProperty("to")
+    private final AddressProposition to;
 
+    @JsonProperty("gasPrice")
+    private final BigInteger gasPrice;
 
-    // depends on the transaction
-    public EthereumTransaction(
-            RawTransaction transaction
-    ) throws NullPointerException {
-        Objects.requireNonNull(transaction, "RawTransaction private data member is null!");
-        if (transaction instanceof SignedRawTransaction)
-            Objects.requireNonNull(((SignedRawTransaction) transaction).getSignatureData(), "signature data can not be null in a signed transaction!");
-        this.transaction = transaction;
-        this.txCost = super.maxCost();
+    private final BigInteger gasLimit;
+    private final BigInteger value;
+    private final Long chainId;
+    private final byte[] data;
+
+    @JsonProperty("maxPriorityFeePerGas")
+    private final BigInteger maxPriorityFeePerGas;
+    @JsonProperty("maxFeePerGas")
+    private final BigInteger maxFeePerGas;
+
+    private final SignatureSecp256k1 signature;
+
+    private AddressProposition from;
+    private String hashString;
+    private BigInteger txCost;
+
+    private synchronized String getTxHash() {
```

Comment:
I think synchronized can be removed, because even if more than 1 thread accesses the same object at the same time there no "corruption". The same for the other synchronized methods

Reply:
let's keep it for safety, same as for all cached values usage we did before

Reply:
Ok